### PR TITLE
do not use >1 threads for velveth in velveloptimizer

### PR DESCRIPTION
**Pull request by bernt-matthias**
velvetoptimizer runs with `-t` already multithreaded (one for each k).
with `OMP_NUM_THREADS` > 1 also each velveth run will be multithreaded.

so overall CPUs are overutilized.

removed `export OMP_THREAD_LIMIT=1` because otherwise the tool gets
stuck.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/velvet_optimiser/velvetoptimiser.xml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
     <description>Automatically optimize Velvet assemblies</description>
     <requirements>
         <requirement type="package" version="1.2.10">velvet</requirement>
         <requirement type="package" version="2.2.6">perl-velvetoptimiser</requirement>
     </requirements>
     <version_command>VelvetOptimiser.pl --version</version_command>
     <command detect_errors="exit_code"><![CDATA[
-        export OMP_NUM_THREADS=2 &&
-        export OMP_THREAD_LIMIT=2 &&
+        export OMP_NUM_THREADS=1 &&
```
Comment (mvdbeek) [inline tools/velvet_optimiser/velvetoptimiser.xml:9]: Shouldn't this be `"\${GALAXY_SLOTS:-1}" or "\${OMP_NUM_THREADS:-1}" ? Don't know if GALAXY_SLOTS is a good proxy, but ideally admins can override the default.

Comment (bernt-matthias) [reply tools/velvet_optimiser/velvetoptimiser.xml:9]: There are two levels of parallelization here. 

1. velveth (which is called internally): the number of threads is controlled via environment variable  `OMP_NUM_THREADS` that needs to be set (I recall that not setting it leads to using max number of CPUs)
2. VelvetOptimiser starts multiple threads (one per value of k) which is controlled via `-t`

Previously both levels of parallelization were used. Leading to overutilization by a factor of 2.

So we can either 

- use `-t "${GALAXY_SLOTS:-1}"` or and set `OMP_NUM_THREADS` to 1 OR
- the opposite (i.e. `-t 1` and `OMP_NUM_THREADS="${GALAXY_SLOTS:-1}"`)

Comment (mvdbeek) [reply tools/velvet_optimiser/velvetoptimiser.xml:9]: I see. I guess in this case it makes more sense to do what you're doing, but maybe make the `OMP_NUM_THREADS ` / `OMP_THREAD_LIMIT ` configurable with a default of 1 ? And maybe add this as a comment, that may come in handy in the future.

Comment (bernt-matthias) [reply tools/velvet_optimiser/velvetoptimiser.xml:9]: Comment is no problem.

Configurable by the user? Or do you have something else in mind?

Comment (Slugger70) [reply tools/velvet_optimiser/velvetoptimiser.xml:9]: Hey @bernt-matthias, I actually use the OMP_NUM_THREADS when we use VelvetOptimiser (though we only use it for training as it is a very old assembler with much better and newer ones available). So I agree with Marius, I'd prefer it to be configurable by the user.

Comment (bernt-matthias) [reply tools/velvet_optimiser/velvetoptimiser.xml]: OK. Did so as an advanced option.

**Comment by bgruening** [comment]
Offering this to a user is not a good idea imho. No one knows what to do with that, isn't it?

**Comment by bernt-matthias** [comment]
> Offering this to a user is not a good idea imho.

I also think so. I think we should decide for one of the two or we need to balance, e.g. using two threads for each velvet call and use `-t GALAXY_SLOTS / 2`

File: tools/velvet_optimiser/velvetoptimiser.xml (hunk 1 above)
Comment (bernt-matthias) [reply tools/velvet_optimiser/velvetoptimiser.xml]: @Slugger70 would it be fine with you to fix `OMP_NUM_THREADS` to one. You would still be able to use multiple threads (i.e. multiple parallel runs of velvet). 

Alternatively I would suggest that I add a check if OMP_NUM_THREADS is set and if not set it to one. Thereby the admin would be able to set it via an environment variable.

**Comment by bernt-matthias** [comment]
Anyone wants to decide on this one?

**Review by bgruening — APPROVED** [review]

**Comment by bernt-matthias** [comment]
I'm going to merge this if nobody objects

**Pull request merged**